tmva/tmva/CMakeLists.txt

```diff
@@ -166,8 +166,16 @@ set(TMVA_HEADERS
   TMVA/VarTransformHandler.h
   TMVA/Version.h
   TMVA/Volume.h
+  TMVA/RTensor.hxx
+  TMVA/RTensorUtils.hxx
 )
 
+if(root7)
+  list(APPEND TMVA_HEADERS
+      TMVA/RTensor.hxx
+      TMVA/RTensorUtils.hxx)
+endif()
```

Comment:
I would make this part of regular ROOT unless it needs C++14. I can see how RDataFrame will make use of this and break if it's not enabled. For a similar reason this might be better located somewhere in `math/` and not `tmva/` since TMVA can be disabled.

Reply:
How would you address the dependency of `RTensorUtils.hxx` on RDataFrame? We cannot put it in VecOps but as well, forcing all TMVA users to build RDataFrame is not the solution. Since it's experimental (and I don't want to introduce a new cmake flag), I would have reused the root7 flag. As well, I expect that we depend on c++14 in the future.

Reply:
And as long as RDataFrame does not depend, I would keep RTensor in TMVA. Since it's header-only, moving it to VecOps is trivial.

Reply:
There is no option to disable VecOps or RDataFrame in ROOT, but there is to disable TMVA or ROOT 7. If moving to VecOps is feasible, then it's probably best to be there, along with RVec. The Utils header can be somewhere else, though. Maybe @dpiparo or @bluehood have better ideas about how to organize the code.

Reply:
@amadio RDF will presumably never depend on TMVA (the opposite will be true), and at present there are no plans to use RTensor in RDF (first _TTreeReader_ must support n-dim arrays in some form, which is not happening in the immediate future).

If I understand correctly, the rationale for having TMVA::Experimental in root7 is that otherwise _current_ libTMVA ends up depending on libRDF. @stwunsch maybe that's less of a problem than we thought given that there is no way to disable VecOps and RDF?

Overall, I wouldn't really know whether it's better to have RTensor in TMVA::Experimental or VecOps::Experimental: I don't see a huge difference for now, and given the dependencies between libraries, it can always be moved at a later time if we decide the other way is better.